src/telemetry/net_if_properties.rs

```diff
@@ -0,0 +1,209 @@
+/*
+ * This file is part of Edgehog.
+ *
+ * Copyright 2022 SECO Mind Srl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use crate::error::DeviceManagerError;
+use astarte_sdk::types::AstarteType;
+use log::warn;
+use std::collections::HashMap;
+
+#[derive(Debug)]
+enum TechnologyType {
+    Ethernet,
+    Cellular,
+    WiFi,
+}
+
+impl ToString for TechnologyType {
+    fn to_string(&self) -> String {
+        match self {
+            TechnologyType::Ethernet => "Ethernet",
+            TechnologyType::Cellular => "Cellular",
+            TechnologyType::WiFi => "WiFi",
+        }
+        .to_owned()
+    }
+}
+
+#[derive(Debug)]
+struct NetworkInterfaceProperties {
+    interface: String,
+    mac_address: String,
+    technology_type: TechnologyType,
+}
+
+fn get_ethernet_wifi() -> Result<Vec<NetworkInterfaceProperties>, DeviceManagerError> {
```

Comment:
This actually returns also cellular, so I'd rather call it `get_supported_network_interfaces`